cloud/wfm/src/executor/index.ts

```diff
@@ -0,0 +1,45 @@
+import * as Promise from 'bluebird';
+import { Process } from '../process';
+import { ProcessInstance } from '../process-instance';
+import { Task, TaskEventData, TaskStatus } from '../task';
+
+/**
+ * Executor engine for a Process
+ *
+ * Triggers instantiation and execution of a ProcessInstance
+ */
+export interface Executor {
+  process: Process;
+  instance?: ProcessInstance;
+  start(): void;
+}
+
+export interface InstanceRepository {
+  save(instance: ProcessInstance): Promise<ProcessInstance>;
+}
+
+class ExecutorImpl implements Executor {
+  public instance: ProcessInstance;
+  constructor(public process: Process, public instanceRepository: InstanceRepository) {
+    this.instance = this.process.createInstance();
+  }
+  public start() {
+    this.runCurrentTask();
+  }
+  protected onTaskDone(e: TaskEventData<Task>) {
+    if (e.currentStatus === TaskStatus.done) {
+      Promise.resolve(this.instance.next())
+        .then(() => this.saveInstance)
```

Comment:
the problem with passing the function is that it's not bound to `this`, so the equivalent is `.then(this.saveInstance.bind(this))`, though since we have arrow functions I think it's more readable with them.